app/src/features/projects/components/ProjectFundingForm.test.tsx

```diff
@@ -95,4 +95,152 @@ describe('ProjectFundingForm', () => {
+      funding_sources: [
+        {
+          id: 11,
+          agency_id: 1,
+          investment_action_category: 1,
+          investment_action_category_name: 'action 1',
+          agency_project_id: '111',
+          funding_amount: 222,
+          start_date: '2021-03-14',
+          end_date: '2021-04-14',
+          revision_count: 23
+        },
+        {
+          id: 12,
+          agency_id: 2,
+          investment_action_category: 2,
+          investment_action_category_name: 'category 1',
+          agency_project_id: '112',
+          funding_amount: 223,
+          start_date: '2021-03-15',
+          end_date: '2021-04-15',
+          revision_count: 24
+        },
+        {
+          id: 13,
+          agency_id: 3,
+          investment_action_category: 3,
+          investment_action_category_name: 'not applicable',
+          agency_project_id: '113',
+          funding_amount: 224,
+          start_date: '2021-03-16',
+          end_date: '2021-04-16',
+          revision_count: 0
+        }
+      ]
+    };
+
+    const { asFragment, getByTestId, queryByText } = render(
+      <Formik
+        initialValues={existingFormValues}
+        validationSchema={ProjectFundingFormYupSchema}
+        validateOnBlur={true}
+        validateOnChange={false}
+        onSubmit={async () => {}}>
+        {() => <ProjectStepComponents component="ProjectFunding" codes={codes} />}
+      </Formik>
+    );
+
+    const addButton = await getByTestId('add-button');
+    expect(addButton).toBeInTheDocument;
+    fireEvent.click(addButton);
+
+    expect(await queryByText('Agency Details')).toBeInTheDocument;
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('shows edit funding source dialog on edit click', async () => {
+    await act(async () => {
+      const existingFormValues: IProjectFundingForm = {
+        funding_sources: [
+          {
+            id: 11,
+            agency_id: 1,
+            investment_action_category: 1,
+            investment_action_category_name: 'action 1',
+            agency_project_id: '111',
+            funding_amount: 222,
+            start_date: '2021-03-14',
+            end_date: '2021-04-14',
+            revision_count: 23
+          }
+        ]
+      };
+
+      const { asFragment, getByTestId, getByText, queryByText } = render(
+        <Formik
+          initialValues={existingFormValues}
+          validationSchema={ProjectFundingFormYupSchema}
+          validateOnBlur={true}
+          validateOnChange={false}
+          onSubmit={async () => {}}>
+          {() => <ProjectStepComponents component="ProjectFunding" codes={codes} />}
+        </Formik>
+      );
+
+      const editButton = await getByTestId('edit-button-0');
+      expect(editButton).toBeInTheDocument;
+
+      fireEvent.click(editButton);
+      expect(await queryByText('Agency Details')).toBeInTheDocument;
+
+      expect(asFragment()).toMatchSnapshot();
```

Comment:
I dont think this snapshot is actually working the way you expect
I would just remove it as I think the other assertions you have in place are sufficient for good quality tests :) or we need to fix this

Reply:
I went through the test thoroughly, and I'm pretty sure the test is doing what I want it to do.  Let's talk if you think I'm missing something.

Reply:
I don't see the dialog in the snapshot.  I tried checking it out locally and tweaking a few things that I thought would help make it show up, but It still didn't seem to show up, despite the related expects on the title passing and the test itself passing.

It looks like everything is working as expected, but probably the fragment just doesn't encapsulate the dialog (since it renders at a higher level in the DOM).  Possibly we would need to use `baseElement` instead of `asFragment`?

In any case, the actual expects and clicks, etc, are working fine. So we can either update the snapshot check to use `baseElement` (if my above guess is correct), or just remove the snapshot check as the other expects are already correctly testing the flow.